dags/seizure_detection_pipeline.py

```diff
@@ -160,35 +192,44 @@ def t_compute_consolidate_feats_and_annot(parameters: dict) -> dict:
 #        }
 #    )
 
-    df_db = pd.read_csv(f'{FETCHED_DATA_FOLDER}/df_candidates.csv', encoding='utf-8')
+    def get_initial_parameters(df_db: pd.DataFrame) -> List[dict]:
+        parameters_list = []
+        for index in range(df_db.shape[0]):
+            qrs_file_path = df_db['edf_file_path'].iloc[index]
+            tse_bi_file_path = df_db['annotations_file_path'].iloc[index]
+            exam_id = df_db['exam_id'].iloc[index]
+            parameters = {'qrs_file_path': ''.join([AIRFLOW_PREFIX_TO_DATA,
+                                                    qrs_file_path]),
+                            'annotations_file_path': ''.join([AIRFLOW_PREFIX_TO_DATA,
+                                                            tse_bi_file_path]),
+                            'exam_id': exam_id,
+                            'method': DETECT_QRS_METHOD,
+                            'rr_intervals_folder': RR_INTERVALS_FOLDER,
+                            'features_folder': FEATURES_FOLDER,
+                            'consolidated_folder': CONSOLIDATED_FOLDER}
+            parameters_list.append(parameters)
+        return parameters_list
 
-    for index in range(df_db.shape[0]):
-        qrs_file_path = df_db['edf_file_path'].iloc[index]
-        tse_bi_file_path = df_db['annotations_file_path'].iloc[index]
-        exam_id = df_db['exam_id'].iloc[index]
-        parameters = {'qrs_file_path': ''.join([AIRFLOW_PREFIX_TO_DATA,
-                                                qrs_file_path]),
-                      'annotations_file_path': ''.join([AIRFLOW_PREFIX_TO_DATA,
-                                                        tse_bi_file_path]),
-                      'exam_id': exam_id,
-                      'method': DETECT_QRS_METHOD,
-                      'rr_intervals_folder': RR_INTERVALS_FOLDER,
-                      'features_folder': FEATURES_FOLDER,
-                      'consolidated_folder': CONSOLIDATED_FOLDER}
-
-        qrs_parameters = t_detect_qrs(parameters)
-        parameters = t_consolidate_parameters(parameters,
-                                              qrs_parameters)
-
-        features_parameters = t_compute_hrv_analysis_features(parameters)
-        parameters = t_consolidate_parameters(parameters,
-                                              features_parameters)
-
-        consolidated_feats_and_annot_parameters = \
-            t_compute_consolidate_feats_and_annot(parameters)
-        parameters = t_consolidate_parameters(
-            parameters,
-            consolidated_feats_and_annot_parameters)
+    df_db = pd.read_csv(f'{FETCHED_DATA_FOLDER}/df_candidates.csv', encoding='utf-8')
+    partition_size = (df_db.shape[0] + NUM_PARTITIONS - 1) // NUM_PARTITIONS
+    for i in range(NUM_PARTITIONS):
+        start = i * partition_size
+        end = (i+1) * partition_size
+        parameters_list = get_initial_parameters(df_db.iloc[start:end])
```

Comment:
Please change i for more explicit variable for easier understanding, and also maybe start/end. No problem on the logic.

Reply:
Ok for me in [84d75c1]